app/services/hyrax/workflow/sipity_actions_generator.rb

```diff
@@ -38,11 +39,39 @@ def call
       private
 
         def generate_state_diagram!
+          # look for old states that need to be removed and validate the states can be removed.
+          removed_states = validate_states_to_be_removed
+
+          # remove any actions that will no longer be needed by the workflow
+          remove_unused_actions
```

Comment:
We aren't :100: consistent about this convention, but I wonder if you would consider making this a `!` method since we don't care about the return value.